signup/models.py

```diff
@@ -56,21 +55,38 @@ def create_inactive_user(self, email, **kwargs):
         login. Our definition of inactive is thus a user that has an invalid
         password.
         """
+        username_max_length = self.model._meta.get_field('username').max_length
         username = kwargs.pop('username', None)
         if not username:
-            username = email.split('@')[0] \
-                + ''.join(random.choice('0123456789') for count in range(3))
-        user = self.create_user(username, email=email, **kwargs)
-
-        # Force is_active to True and create an email verification key
-        # (see above definition of active user).
-        user.is_active = True
-        salt = hashlib.sha1(str(random.random())).hexdigest()[:5]
-        if isinstance(username, unicode):
-            username = username.encode('utf-8')
-        user.email_verification_key = hashlib.sha1(salt+username).hexdigest()
-        user.save()
-        return user
+            username = email
+        if '@' in username:
+            username = slugify(username.split('@')[0])
+        if len(username) > username_max_length:
+            username = username[:username_max_length]
+
+        username_base = username
+        if len(username_base) > username_max_length - 3:
```

Comment:
If, later on, you are going to use:

    suffix = ''.join(random.choice('0123456789') for count in range(3))
    username = "%s*-%s*" % (username_base, suffix)

Then the count should be 4 not 3 because the '-' character should be counted.